sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationSetting.cs

```diff
@@ -11,7 +11,7 @@ namespace Azure.Data.AppConfiguration
     /// <summary>
     /// A setting, defined by a unique combination of a key and label.
     /// </summary>
-    public sealed class ConfigurationSetting : IEquatable<ConfigurationSetting>
+    public sealed class ConfigurationSetting
```

Comment:
Why make this change, though? Looking at the bug, I don't see why this helps. `EqualityComparer<T>.Default` would use this implementation of `IEquatable<T>` anyway. This doesn't seem to add any value, and is inconsistent with what we do with "extensible enums".

Reply:
@heaths, could you say more about what you do with extensible enums?

Reply:
We just implement `IEquatable<T>`. This way, the default comparer that many methods use just use that implementation, and you don't have to pass in a comparer to get the right behavior, i.e. "It Just Works". 🙂

Reply:
Per offline discussion, we will not implement IEquatable for ConfigurationSetting.  We will look into providing verbosely named Comparers after GA.

Reply:
So, should I merge this PR?

Reply:
Yep!